oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/NodeStateEntryWriter.java

```diff
@@ -118,14 +122,53 @@ public static String getPath(String entryLine) {
         return entryLine.substring(0, getDelimiterPosition(entryLine));
     }
 
+    /*
+        Currently we only have 2 types of store:
+        1. FlatFileStore
+        2. IncrementalFlatFileStore
+     */
     public static String[] getParts(String line) {
```

Comment:
Can we create a json encompassing the fields  4 fields listed `path|nodeData|checkpoint|operand`?  This will make the code more readable as the whole parsing and writing is a little hacky and will have to change again to parse postions if we need to add more fields.

- It will increase the size of the field and file a little but storage requirements can be managed by compressing it (we should get good compression).
- Parsing we can manage by just reading json at 1 level and not parsing the node data.

Reply:
Yes, it will be a better fit with structure like json. I think it would make sense to discuss this and do it in a separate ticket.